lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -3474,38 +3493,52 @@ void replaceScalarArrayWithVectorArray(Value *ScalarArray, Value *VectorArray,
     if (U->user_empty())
       continue;
     if (BitCastInst *BCI = dyn_cast<BitCastInst>(U)) {
-      BCI->setOperand(0, VectorArray);
+      // Avoid replacing the dest of the memcpy to support partial replacement.
+      if (MC->getArgOperand(0) != BCI)
```

Comment:
Should there be a nullptr check for `MC` ?

Reply:
No.
The MC is from lowerMemcpy which start when have memcpy.